ledger/catchpointtracker.go

```diff
@@ -932,6 +932,11 @@ func (ct *catchpointTracker) postCommitUnlocked(ctx context.Context, dcc *deferr
 // Tracker might update own state in this case. For example, account catchpoint tracker cancels
 // scheduled catchpoint writing that deferred commit.
 func (ct *catchpointTracker) handleUnorderedCommitOrError(dcc *deferredCommitContext) {
```

Comment:
this is not quite right, there are two places where the function is called, and one of them is un-ordered commits. In this case we do not want to reset the trie

Reply:
Good call, thought about this over the weekend -- ended up splitting out this function to be 1:1 with the stage of the commit they're handling. Now this clear-out only happens on commitError specifically.